### ISSUE TITLE
Brice theme demo content header needs alignment removed for RTL

### ISSUE THREAD

**Issue by lsl** [Automattic/wp-calypso#49242]
`.has-text-align-left` needs to be removed from the header example content of the Brice theme.

![Screenshot_2021-01-25 WordPress com](https://user-images.githubusercontent.com/811776/105657140-dc4f4080-5f17-11eb-8ea0-100c1e08bdb4.png)

The theme is supporting RTL correctly but the title that shows up from demo content has left alignment applied, removing this is probably all that is needed to fix things for RTL (this time)

@andrewserong I'm 50-50 on whether PTK should handle alignment replacement as part of the block translation, the more I think about it the more I believe it's in scope. 

Would want to take this on or should gardeners(@p-jackson)/ganon look at doing this in the demo endpoint separate from PTK as part of headstart?

### Steps to reproduce the behavior

Select brice theme under /new onboarding, launching the site should result in something like this: https://test523557425.wordpress.com/

**Comment by p-jackson** [comment]
> I'm 50-50 on whether PTK should handle alignment replacement as part of the block translation

This sounds right to me. Long term it seems like something phase 3 Gutenberg would handle, which means short/medium term it'd be appropriate for the Pattern Toolkit IMO.

**Comment by andrewserong** [comment]
> I'm 50-50 on whether PTK should handle alignment replacement as part of the block translation, the more I think about it the more I believe it's in scope.

I'm 50/50 on this too — I agree it's in scope for what the PTK _should_ be able to handle, but at the same time there are potentially a lot of edge cases to handle. I wonder what the most minimal version of this would be for common things in patterns? Would doing headings and paragraphs be enough to begin with? Re-ordering columns in a reliable way could be very tricky.

One concern is that it would add additional complexity to the parsers, potentially making them brittle, because it'd need to update both the block comment `<!-- wp:heading {"textAlign":"left"} -->` and the class within the markup, e.g. `<h2 class="has-text-align-left">` to become `<h2 class="has-text-align-right">`. At the moment the parsers don't know which language they're dealing with, they just parse content and return an updated block. To keep it simple, perhaps this is just a matter of passing in an `is_rtl` flag to the parsers, and then we could add it in as we go to the specific parsers that need it? That might be the lowest effort way to add it in.

> The theme is supporting RTL correctly but the title that shows up from demo content has left alignment applied, removing this is probably all that is needed to fix things for RTL (this time)

If we can work around it on a case-by-case basis in the short term, that'd be great. But I see what you mean, I can imagine this becoming more of an issue as we go along.

> Would want to take this on or should gardeners(@p-jackson)/ganon look at doing this in the demo endpoint separate from PTK as part of headstart?

I think it'd be good to try to tackle this in PTK directly rather than adding adding additional handling in the demo endpoint to deal with it. If Ganon have scope to work on it, I'm happy to help with reviews, otherwise I can add it to the View backlog, but I'm not sure yet where it'd sit in priorities. If you have specific thoughts or ideas of how it should work, I'm happy to write up an issue for it 🙂

**Comment by lsl** [comment]
Do the parsers expose classes yet?

I'm thinking this could just be part of pattern translation. But that would need the class extract/replace functionality in the more general patterns code.

---

Your other remarks make me think the right move here is actually just to remove the alignment off the source site. That is, it doesn't make sense to be reversing design/structure (that's what the RTL css does or is supposed to do.)

**Comment by andrewserong** [comment]
> Do the parsers expose classes yet?

Not really, there are a couple of parsers that search for a class in the DOM to make sure they're updating the correct node, but we're not updating any class names yet. I think we'd also want to be pretty careful if we wind up replacing things in class names due to how frequently they seem to change / get deprecated for particular blocks. Fortunately, we seem to have managed pretty well so far without being caught by deprecations! 🤞

> the right move here is actually just to remove the alignment off the source site

I agree. It'd be interesting to see how much we can get away with by avoiding using left/right alignment within patterns directly. If we reach a point where it looks like there's a whole category of patterns / layouts that really need alignment it might make it easier to justify the extra work to add in the complexity.

**Comment by simison** [comment]
@andrewserong @ianstewart this seems better handled in View team? Lemme know if not though!

**Comment by andrewserong** [comment]
Thanks @simison, yep, I think that sounds fine for us to do!

**Comment by worldomonation** [comment]
Hi! I'm performing triage on older bugs as part of Quality Squad's efforts to cut down on backlog issue counts.

I was able to reproduce this issue on Firefox 94.0a1 and accessing Gutenboarding, then changing language to Hebrew.

![Screen Shot 2021-10-04 at 4 31 42 PM](https://user-images.githubusercontent.com/6549265/135937997-fd25b353-112e-4978-99a3-1cb4b486bd2e.png)
